sql-plugin/src/main/scala/org/apache/spark/sql/rapids/datetimeExpressions.scala

```diff
@@ -892,8 +909,20 @@ abstract class GpuToTimestampImproved extends GpuToTimestamp {
           failOnError)
       }
     } else if (lhs.dataType() == DateType){
-      lhs.getBase.asTimestampSeconds()
-    } else { // Timestamp
+      timeZoneId match {
```

Comment:
Use `zoneId` defined in `TimeZoneAwareExpression`
```
@transient lazy val zoneId: ZoneId = DateTimeUtils.getZoneId(timeZoneId.get)
```

Reply:
ditto